RHDEVDOCS-4340-document-scrape-profiles-in-CMO (#56073)
openshift/openshift-docs: modules/monitoring-configuring-metrics-collection-profiles.adoc

// Module included in the following assemblies:
//
// * monitoring/configuring-the-monitoring-stack.adoc

:_content-type: CONCEPT
[id="configuring-metrics-collection-profiles_{context}"]
= Configuring metrics collection profiles
 
[IMPORTANT]
====
[subs="attributes+"]
Metrics collection profiles are a Technology Preview feature only. Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete.
Red Hat does not recommend using them in production. 
These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.

For more information about the support scope of Red Hat Technology Preview features, see link:https://access.redhat.com/support/offerings/techpreview[https://access.redhat.com/support/offerings/techpreview].
====

By default, Prometheus collects a large amount of time-series metrics data from default metrics targets.

> Using a metrics collection profile for core {product-title} monitoring components, you can choose to collect either the default amount of metrics data or a minimal amount of data. 

Reviewer: Reading this to me seems a bit that by default we are collecting too many metrics. Let me know if by change you also feel the same or it's just me. 

Maybe I'm more on the side of not saying we collect a large number of metrics but rather that we collect all the metrics exposed by the default targes. However, in some cases:
- a clusters reach a big enough size where those default metrics demand users to run bigger prom instances
- users only care about the platform alerts and never look at the other metrics 

Is that we can discard some metrics, and should use a metric collection profile.

Author: That's a good point. I think I need to reword it.

Reviewer: Much better IMO! Feel free to mark this as resolved

By choosing to collect a minimal amount of metrics data, you can lower the amount of CPU and memory resources used by the monitoring stack while simultaneously maintaining basic monitoring features such as alerting.

[id="about-metrics-collection-profiles_{context}"]
== About metrics collection profiles

You can choose one of two metrics collection profiles:

* *Full*: Prometheus collects all metrics data exposed by the different platform components.

> * *Minimal*: Prometheus only collects the metrics data required for platform alerts, recording rules, telemetry and console dashboards.

Reviewer: Maybe we should add something saying that when using a profile other than "Full" some glitches might show up at the console level, but that no metrics used in alerts are ever discarded.
cc @simonpasquier

Author: If we include information like this, it should be at the most generic level. But I'd prefer not to include anything about console error messages unless absolutely necessary. That said, I'll defer to @simonpasquier's guidance here.

Author: @simonpasquier What's your take on including a note that console errors or glitchiness might occur, but no metrics used in alerts are ever discarded? My preference is to leave information like this out of the core docs and include it as a known issue in the release notes if necessary. WDYT?

Author: I'm going to make a call and say that any user-facing console error messages or bugs should be reported as known issues in the Release Notes. This is a TP feature, so users are already warned that the feature might have some glitches. So I am setting this to resolved.

// Module included in the following assemblies:
//
// * monitoring/configuring-the-monitoring-stack.adoc

:_content-type: CONCEPT
[id="configuring-metrics-collection-profiles_{context}"]

> = Configuring metrics collection profiles

Reviewer 2: **Suggestion**: Avoid using gerunds in concept titles. Refer to: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#writing-concepts and https://redhat-documentation.github.io/modular-docs/#creating-concept-modules.

Author: That's a good point and good for you to raise it here, but because the feature is called "metrics collection profiles" I think it's best to use that phrase in the title, so I'm going to leave it as-is

Reviewer 2: @bburt-rh I suggest the heading as: **Configuration of  metrics collection profiles**

That is more suitable here. I trust you to consider and implement it. WIth that, I approve the PR.

 
[IMPORTANT]
====
[subs="attributes+"]
Using a metrics collection profile is a Technology Preview feature only. Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete.
Red Hat does not recommend using them in production. 
These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.

For more information about the support scope of Red Hat Technology Preview features, see link:https://access.redhat.com/support/offerings/techpreview[https://access.redhat.com/support/offerings/techpreview].
====

> By default, Prometheus collects all metrics exposed by the default metrics targets in {product-title} monitoring components. 

Reviewer: Adding this because even in the full profile some filtering was already being done
Also removing "monitoring" since Prometheus doesn't only collect metrics for monitoring components but all in OpenShift
```suggestion
By default, Prometheus collects all desired metrics exposed by the default metrics targets in {product-title} components. 
```

> However, you might want Prometheus to collect fewer metrics for a cluster in certain scenarios:

Reviewer: shouldn't it be from?

```suggestion
However, you might want Prometheus to collect fewer metrics from a cluster in certain scenarios:
```

Author: Yes. Correct. Fixed.

> * Users of a cluster require only platform alerts and have no need for other metrics.

Reviewer: I'm not sure "Users" here is the correct term as the only entity that will care about these metrics that we are discussing is the cluster admins. Smth like this
```suggestion
* Cluster admin requires only alert and console metrics and does not require other metrics to be available.
```

Author: Revised per your suggestion.

> * If cluster administrators require only alert and console metrics and do not require other metrics to be available.

Reviewer 3: missed `telemetry` according to https://github.com/openshift/openshift-docs/pull/56073#discussion_r1123610620

Author: Added per your suggestion.

> * If a cluster increases in size, and the increased size of the default metrics data being collected now requires increased CPU and memory resources. 

Reviewer: A bigger cluster will always request more CPU and memory, but feel free to ignore this suggestion if you feel like it doesn't contribute 
```suggestion
* If a cluster increases in size, and the increased size of the default metrics data being collected now requires an unreasonable increase of CPU and memory resources. 
```

Author: Revised to say `now requires a significant increase in CPU and memory resources`

> * If a cluster increases in size, and the increased size of the default metrics data being collected now requires a significant increase in CPU and memory resources. 

Author: Good suggestions. Changed per your edits.

> You can use a metrics collection profile to choose to collect either the default amount of metrics data or a minimal amount of metrics data. 

Author: Changed per your suggestion.

> When you choose to collect a minimal amount of metrics data, basic monitoring features such as alerting continue to work.

Reviewer 2: ```suggestion
When you collect minimal metrics data, basic monitoring features continue to work.
```

Author: For concision, I changed the first part the sentence per your suggestion, but I'm leaving in the reference to alerting, because users need to know that alerting remains available.

At the same time, the CPU and memory resources required by Prometheus are lowered.

[id="about-metrics-collection-profiles_{context}"]
== About metrics collection profiles

You can enable one of two metrics collection profiles:

> * *Full*: Prometheus collects all metrics data exposed by the different platform components. This setting is the default.

Reviewer 3: from https://github.com/openshift/cluster-monitoring-operator/blob/0157b7d01e288eb26ae6e220295b529740902fd3/pkg/manifests/types.go#L24-L27
since the valid value is full,minimal, we should use the same values here

Author: fixed

> * *full*: Prometheus collects all metrics data exposed by the different platform components. This setting is the default.

Reviewer: Adding this because even in the full profile some filtering was already being done
```suggestion
* *full*: Prometheus collects all desired metrics data exposed by the different platform components. This setting is the default.
```

> * *minimal*: Prometheus collects only the metrics data required for platform alerts, recording rules, telemetry, and console dashboards.

Reviewer 2: @bburt-rh The 2nd option is this: "Prometheus collects only the metrics data required for platform alerts, recording rules, and telemetry and console dashboards."

Please recheck and whatever is accurate technically, accordingly modify the sentence.

Author: Sorry. My misunderstanding. The technically correct phrasing is ``platform alerts, recording rules, telemetry, and console dashboards`. So leaving as-is. Thanks

Reviewer 3: yes, no need to change, `platform alerts, recording rules, telemetry, and console dashboards` is right

Reviewer 2: @bburt-rh @juzhao Thanks for the confirmation.

> * *minimal*: Prometheus collects only the metrics data required for platform alerts, recording rules, telemetry and console dashboards.

Author: The subheading with this content is in the concept file, and not in the procedure file, so no changes are required. Setting as resolved.

> * *minimal*: Prometheus collects only the metrics data required for platform alerts, recording rules, telemetry and console dashboards.

Reviewer 2: @bburt-rh my bad. Did not see properly. sorry about that.
